### DISCUSSION TITLE
how to implement line breaks

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-langium/langium#1731] [Q&A] [answered]
how to implement line breaks("\n) by using SignatureInformation releated API
i  want to achieve like 
![image](https://github.com/user-attachments/assets/36d0df3b-befc-4704-bde8-df6e9e3bad21)
but now i can only arrange in one Line
![image](https://github.com/user-attachments/assets/0f0f7710-0b97-4b10-bba3-23ae40119aab)

**Comment by zrwsmd** [comment]
i extends AbstractSignatureHelpProvider,and override getSignatureFromElement method
my code is below ,i add '\n',but didn't work out well
![image](https://github.com/user-attachments/assets/5e08d9fc-2b75-4d28-814a-6928c2882d88)

**Reply by msujew** [reply, marked as answer]
Hey @zrwsmd,

thanks for the snippet, that helps a lot in figuring out the problem. The label cannot contain newlines. Instead, you should use the `documentation` property of the `SignatureInformation` type to display formatted data. You should create a `MarkupContent` object for it to be rendered as markdown.

**Comment by zrwsmd** [comment]
thanks a lot ,it works!
the following code can be referd to
wrapped  with **  ** can bold the code
italic code can be wrapped with  *  *
![image](https://github.com/user-attachments/assets/8abe778a-8c92-4154-8985-9376d319509c)
![image](https://github.com/user-attachments/assets/967057b8-fb06-41ac-818c-4aaed8a817c3)

**Reply by zrwsmd** [reply]
everything is ok,but it is weird that this two string have same length,but can't alignment
![image](https://github.com/user-attachments/assets/7917dfa5-a71c-4562-b252-60435ff50322)

**Comment by zrwsmd** [comment]
i want to achieve like this
![image](https://github.com/user-attachments/assets/113e81cb-b1fd-4f3f-b12b-0bdd1df18366)

**Reply by msujew** [reply]
I'm not entirely sure, whether vs code supports this, but you might want to try using a markdown table. Alternatively, you can wrap everything in a ```` ``` ```` block to ensure that the output is in a monospace font.

**Reply by zrwsmd** [reply]
thanks for your tip,it works 
![image](https://github.com/user-attachments/assets/491e3fa8-e739-4328-b081-41028cd191c8)

**Comment by zrwsmd** [comment]
![image](https://github.com/user-attachments/assets/ea01602b-6609-491d-ac48-0491ad069276)
